Dockerfile

```diff
@@ -0,0 +1,23 @@
+ARG PYTHON_VERSION=3.10
+
+FROM python:${PYTHON_VERSION}-slim AS builder
+
+WORKDIR /app
+
+COPY . .
```

Comment:
Using `COPY . .` can lead to unnecessary files being copied into the Docker image, increasing its size. Consider using a `.dockerignore` file to exclude files that are not needed in the image.